app/stac_api/pagination.py

```diff
@@ -0,0 +1,150 @@
+import logging
+from urllib import parse
+
+from django.conf import settings
+from django.core.handlers.wsgi import WSGIRequest
+from django.utils.translation import gettext_lazy as _
+
+from rest_framework import pagination
+from rest_framework.exceptions import ValidationError
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.utils.urls import remove_query_param
+
+from stac_api.utils import get_query_param
+
+logger = logging.getLogger(__name__)
+
+
+class CursorPagination(pagination.CursorPagination):
+    ordering = 'id'
+    page_size_query_param = 'limit'
+    max_page_size = settings.REST_FRAMEWORK['PAGE_SIZE_LIMIT']
+
+    def get_next_link(self, request=None):  # pylint: disable=arguments-differ
+        next_page = super().get_next_link()
+        if next_page:
+            return {'rel': 'next', 'href': next_page}
+        return None
+
+    def get_previous_link(self, request=None):  # pylint: disable=arguments-differ
+        previous_page = super().get_previous_link()
+        if previous_page:
+            return {'rel': 'previous', 'href': previous_page}
+        return None
+
+    def get_paginated_response(self, data, request=None):  # pylint: disable=arguments-differ
+        links = []
+        next_link = self.get_next_link(request)
+        previous_link = self.get_previous_link(request)
+        if next_link is not None:
+            links.append(next_link)
+        if previous_link is not None:
+            links.append(previous_link)
+
+        if 'links' not in data and not links:
+            data.update({'links': []})
+        elif 'links' not in data and links:
+            data.update({'links': [links]})
+        elif links:
+            data['links'] += links
+        return Response(data)
+
+    def get_page_size(self, request):
+        # Overwrite the default implementation about the page size as this one
+        # don't validate the query parameter, its simply correct it if it is not valid
+        # here we want to return a 400 BAD REQUEST when the provided page size is invalid.
+
+        integer_string = self.get_raw_page_size(request)
+
+        try:
+            page_size = int(integer_string)
+        except ValueError as error:
+            logger.error(
+                'Invalid query parameter limit=%s: must be an integer',
+                integer_string,
+                extra={'request': request}
+            )
+            raise ValidationError(
+                _('invalid limit query parameter: must be an integer'),
+                code='limit'
+            )
+
+        if page_size <= 0:
+            logger.error(
+                'Invalid query parameter limit=%d: negative number not allowed',
+                page_size,
+                extra={'request': request}
+            )
+            raise ValidationError(
+                _('limit query parameter to small, must be in range 1..%d') % (self.max_page_size),
```

Comment:
Its a copy&pasted message here, I know, but being picky here, we could use the chance and correct `to small` to `too small"